src/mem/epoch/markable_atomic.rs

```diff
@@ -5,33 +5,38 @@ use std::sync::atomic::{self, Ordering};
 
 use super::{Owned, Shared, Guard};
 
-/// Like `std::sync::atomic::AtomicPtr`.
+/// Provides atomic access to a `(*const T, usize)` pair.
 ///
-/// Provides atomic access to a (nullable) pointer of type `T`, interfacing with
-/// the `Owned` and `Shared` types.
+/// The pair consists of:
+///
+/// - a (nullable) pointer of type `T`, interfacing with the `Owned` and `Shared` types
+///
+/// - a `usize` marker value that can fit into the unused bits of the pointer:
+/// `mark < std::mem::align_of::<T>()`
 #[derive(Debug)]
-pub struct MarkedAtomic<T> {
+pub struct MarkableAtomic<T> {
```

Comment:
I changed the name to Markable rather than Marked, since the terminology in concurrency papers is often "marked" for 1, "unmarked" for 0, so the value isn't always "marked".